mobile/scripts/buildApp.sh

```diff
@@ -47,84 +54,102 @@ if [[ "${OS}" == "android" ]]; then
   echo "Building for Android 35"
   ANDROID_PLATFORM=android-35
 
-  "$QMAKE_BIN" "$PRO_FILE" "$QMAKE_CONFIG" -spec android-clang ANDROID_ABIS="$ANDROID_ABI" APP_VARIANT="${APP_VARIANT}" VERSION="$DESKTOP_VERSION" -after
+  # Map BUILD_VARIANT to gradle flavor
+  if [[ "$BUILD_VARIANT" == "pr" ]]; then
+    GRADLE_FLAVOR="pr"
+    OUTPUT_NAME="StatusPR"
+  else
+    GRADLE_FLAVOR="production"
+    OUTPUT_NAME="Status"
+  fi
+
+  echo "Using Gradle flavor: $GRADLE_FLAVOR"
+  echo "Target name: $OUTPUT_NAME"
+
+  "$QMAKE_BIN" "$CWD/../wrapperApp/Status.pro" "$QMAKE_CONFIG" -spec android-clang ANDROID_ABIS="$ANDROID_ABI" APP_VARIANT="${APP_VARIANT}" VERSION="$DESKTOP_VERSION" -after
 
   # Build the app
   make -j"$(nproc)" apk_install_target
 
-  if [[ "$BUILD_TYPE" == "aab" ]]; then
-    if [[ -z "$KEYSTORE_PATH" || -z "$KEYSTORE_PASSWORD" || -z "$KEY_ALIAS" || -z "$KEY_PASSWORD" ]]; then
-      echo "Error: AAB builds require signing credentials"
-      echo "Required: KEYSTORE_PATH, KEYSTORE_PASSWORD, KEY_ALIAS, KEY_PASSWORD"
-      exit 1
-    fi
+  # aux-mode copies dependencies AND generates libs.xml
+  echo "Preparing Android build..."
+  androiddeployqt \
+    --input "$BUILD_DIR/android-${OUTPUT_NAME}-deployment-settings.json" \
+    --output "$BUILD_DIR/android-build" \
+    --android-platform "$ANDROID_PLATFORM" \
+    --verbose \
+    --aux-mode
+
+  # Build with gradle directly to use product flavors
+  CUSTOM_ANDROID="$CWD/../android/qt${QT_MAJOR}"
+  ANDROID_BUILD="$BUILD_DIR/android-build"
+
+  # Override with custom Android files (manifest, gradle for flavors, src)
+  # Keep Qt-generated res/ (including libs.xml) but add our custom resources
+  cp "$CUSTOM_ANDROID/AndroidManifest.xml" "$ANDROID_BUILD/"
+  cp "$CUSTOM_ANDROID/build.gradle" "$ANDROID_BUILD/"
+  cp "$CUSTOM_ANDROID/settings.gradle" "$ANDROID_BUILD/"
+  cp "$CUSTOM_ANDROID/gradle.properties" "$ANDROID_BUILD/"
+
+  # Copy custom res files (icons, styles, etc.) but preserve Qt-generated libs.xml
+  if [[ -d "$CUSTOM_ANDROID/res" ]]; then
+    # Copy everything except values/libs.xml (Qt generates that)
+    find "$CUSTOM_ANDROID/res" -type f ! -name "libs.xml" -exec sh -c '
+      src="$1"
+      rel="${src#'"$CUSTOM_ANDROID/res/"'}"
+      dest="'"$ANDROID_BUILD/res/"'$rel"
+      mkdir -p "$(dirname "$dest")"
+      cp "$src" "$dest"
+    ' _ {} \;
+  fi
 
-    if [[ ! -f "$KEYSTORE_PATH" ]]; then
-      echo "Error: Keystore file not found at $KEYSTORE_PATH"
-      exit 1
-    fi
+  if [[ -d "$CUSTOM_ANDROID/src" ]]; then
+    mkdir -p "$ANDROID_BUILD/src"
+    cp -r "$CUSTOM_ANDROID/src/"* "$ANDROID_BUILD/src/"
+  fi
+
+  cd "$BUILD_DIR/android-build"
 
-    echo "Building AAB..."
-    androiddeployqt \
-      --input "$BUILD_DIR/android-Status-deployment-settings.json" \
-      --output "$BUILD_DIR/android-build" \
-      --aab \
-      --release \
-      --android-platform "$ANDROID_PLATFORM"
+  if [[ "$BUILD_TYPE" == "aab" ]]; then
```

Comment:
Branches look very simmilar with very small diff.
I think it should be one code, parametrised:
- file extention: aap, apk
- gradle task: `bundle${GRADLE_FLAVOR^}Release`, `assemble${GRADLE_FLAVOR^}Release`